pkg/controllers/uiplugin/plugin_info_builder.go

```diff
@@ -0,0 +1,110 @@
+package uiplugin
+
+import (
+	"fmt"
+
+	osv1alpha1 "github.com/openshift/api/console/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	uiv1alpha1 "github.com/rhobs/observability-operator/pkg/apis/uiplugin/v1alpha1"
+)
+
+type UIPluginInfo struct {
+	Image             string
+	Name              string
+	ConsoleName       string
+	DisplayName       string
+	Proxies           []osv1alpha1.ConsolePluginProxy
+	Role              *rbacv1.Role
+	RoleBinding       *rbacv1.RoleBinding
+	ResourceNamespace string
+}
+
+func PluginInfoBuilder(plugin *uiv1alpha1.UIPlugin, pluginConf UIPluginsConfiguration, clusterVersion string) (*UIPluginInfo, error) {
+	imageKey, err := getImageKeyForPluginType(plugin.Spec.Type, clusterVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	image := pluginConf.Images[imageKey]
+	if image == "" {
+		return nil, fmt.Errorf("no image provided for plugin type %s with key %s", plugin.Spec.Type, imageKey)
+	}
+
+	name := "observability-ui-" + plugin.Name
+	namespace := pluginConf.ResourcesNamespace
+
+	switch plugin.Spec.Type {
+	case uiv1alpha1.TypeDashboards:
+		{
+			readerRoleName := plugin.Name + "-datasource-reader"
+			datasourcesNamespace := "openshift-config-managed"
```

Comment:
This can be configured in the plugin using the `--dashboards-namespace` flag but its default is `openshift-config-managed`. At this point there is a single deployment template used for any plugin. This might change when we add support for other plugins that need specific options.